### ISSUE TITLE
heroku.account.info wrong number of arguments

### ISSUE THREAD

**Issue by paul** [heroku/platform-api#36]
The docs aren't clear how to get the info for the current account. `#info` (incorrectly, IMHO) requires at least 1 argument, while the [Platform API](https://devcenter.heroku.com/articles/platform-api-reference#account-account-info-1) indicates none is required for `/account`.

Digging into the [generated source](https://github.com/heroku/platform-api/blob/master/lib/platform-api/client.rb#L531), the argument name `account_email_or_account_id_or_account_self` implies that an "account_self" is a permissible option, but it is not at all clear how to get one.

My wild-ass guesses:

```
[1] pry(main)> heroku = PlatformAPI.connect_oauth("...")
=> #<PlatformAPI::Client:0x007fd21e3fd520 @client=#<Heroics::Client url="https://api.heroku.com">>
[2] pry(main)> heroku.account.info
ArgumentError: wrong number of arguments (0 for 1)
from /Users/rando/.gem/ruby/2.2.3/gems/platform-api-0.5.0/lib/platform-api/client.rb:531:in `info'
[3] pry(main)> heroku.account.info nil
Excon::Errors::NotFound: Expected([200, 201, 202, 204, 206, 304]) <=> Actual(404 Not Found)
from /Users/rando/.gem/ruby/2.2.3/gems/excon-0.45.4/lib/excon/middlewares/expects.rb:6:in `response_call'
[4] pry(main)> heroku.account.info "self"
Excon::Errors::NotFound: Expected([200, 201, 202, 204, 206, 304]) <=> Actual(404 Not Found)
from /Users/rando/.gem/ruby/2.2.3/gems/excon-0.45.4/lib/excon/middlewares/expects.rb:6:in `response_call'
[5] pry(main)> heroku.account.info(heroku.account)
Excon::Errors::NotFound: Expected([200, 201, 202, 204, 206, 304]) <=> Actual(404 Not Found)
from /Users/rando/.gem/ruby/2.2.3/gems/excon-0.45.4/lib/excon/middlewares/expects.rb:6:in `response_call'
```

**Comment by paul** [comment]
I tossed a debugger in to see what URLs were being hit, and to my surprise the Excon path is `/users/{whatever I put}`. I dug into the schema.json, and it looks like the account info is being defined in two places, [Line 344](https://github.com/heroku/platform-api/blob/master/schema.json#L343-L352) and then again in the same `links` array on [Line 444](https://github.com/heroku/platform-api/blob/master/schema.json#L343-L352). Is this a bug in the schema dump?

**Comment by geemus** [comment]
This is because there are cases where you want to get info about an account that is not yours, so we implemented it in that more general way, instead of being just your own account. Anyway, self is `~`. If you pass that it will be yourself, the client should perhaps use this value as the default if no value is passed (but I would worry about this leading to unexpected/erroneous values, so maybe explicit is better anyway).

I have no idea why that is duplicated in the links, will have to dig in.

**Comment by paul** [comment]
Ok, that's what I was looking for. Is the `~` argument documented anywhere, I just missed it? Running it with that still makes the request to `/users`, but I get the account back as documented in the Platform API docs. Is `/users` a previous version of the API?

```
[2] pry(main)> heroku.account.info "~"
{
  :path     => "/users/~",
  :root_url => "https://api.heroku.com"
}
=> {"allow_tracking"=>true,
 "beta"=>false,
...
```

Anyways, thanks for your help.

**Comment by geemus** [comment]
@paul it is supposed to be documented in the schema, but looks like the copy of schema in this repo is a little out of date (the updated one includes `~` as part of the identity definition and defines it explicitly). I believe the previous version of the api used `/account` instead of `/users/~`, if that helps. We should get this schema updated, but for now this should hopefully let you keep going at least.

**Comment by paul** [comment]
I curious, then, because the Platform API docs list /accounts, but makes no
mention of /users.


On Mon, Jan 18, 2016 at 8:55 AM, heroku/platform-api
<reply@reply.github.com>
wrote:
Closed #36 [https://github.com/heroku/platform-api/issues/36] .

—
Reply to this email directly or view it on GitHub
[https://github.com/heroku/platform-api/issues/36#event-518073292] .[https://github.com/notifications/beacon/AAAAuIASkUofyyVBlZWUs8A-kKya6CfQks5pbQJ2gaJpZM4HGbFV.gif]

**Comment by geemus** [comment]
Yeah, unfortunately those docs also need an update (it is very near the top of my priority list, but just hasn't quite landed). Hopefully I'll be able to update both later this week though.
